src/utilities/valueFromAST.js

```diff
@@ -93,13 +98,14 @@ export function valueFromAST(
     return Object.keys(fields).reduce((obj, fieldName) => {
       const field = fields[fieldName];
       const fieldAST = fieldASTs[fieldName];
-      let fieldValue =
+      const fieldValue =
         valueFromAST(fieldAST && fieldAST.value, field.type, variables);
-      if (isNullish(fieldValue)) {
-        fieldValue = field.defaultValue;
-      }
-      if (!isNullish(fieldValue)) {
+      if (fieldValue === null) {
+        obj[fieldName] = fieldValue;
+      } else if (!isNullish(fieldValue)) {
         obj[fieldName] = fieldValue;
+      } else if (fieldValue !== null) {
```

Comment:
This could just be `else` since `fieldValue !== null` is guaranteed to be true here, otherwise the first branch would have been followed.

Also, the two conditions above are the same.

I think you can further shorten to a trinary.

```js
// If no valid field value was provided, use the default value
obj[fieldName] = fieldValue === undefined || fieldValue !== fieldValue ? 
  field.defaultValue :
  fieldValue;
```

I think you can probably skip the `isNullish` in favor of inlining the check here. I think `isNullish(x) && x !== null` would be confusing to read.